walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,142 @@
+    TimeRange lastEventTimeRange = eventList.get(eventList.size() - 1).getWhen();
+    
+    if (!availableTimes.isEmpty() && requestFits(request, TimeRange.END_OF_DAY, lastEventTimeRange.end())) {
+      if (nothingEndsTheDay(availableTimes, eventList)) {
+        availableTimes.add(TimeRange.fromStartEnd(lastEventTimeRange.end(),
+          TimeRange.END_OF_DAY, true));
+      } 
+    } else if (fitsOnlyAtTheEnd(request, availableTimes, eventList)) {
+        availableTimes.add(TimeRange.fromStartEnd(lastEventTimeRange.end(),
+          TimeRange.END_OF_DAY, true));
+      }
+      
+
+    return availableTimes;
+  }
+
+  private static boolean requestHasEventAttendees(MeetingRequest request, Event currEvent) {
+    List<String> requestAttendees = new ArrayList<>(request.getAttendees());
+    List<String> eventAttendees = new ArrayList<>(currEvent.getAttendees());
+
+    return requestAttendees.stream().anyMatch(eventAttendees::contains);
+  }
+
+  /**
+   * Check if the first event in our list already includes the start of the day.
+   */ 
+  private static boolean startOfDayIsFree(List<Event> eventList) {
+    return eventList.get(0).getWhen().start() != TimeRange.START_OF_DAY;
+  }
+
+  /**
+   * Check if there are no available times already added in our list which contain the end of day
+   * and any events that end the day.
+   */ 
+  private static boolean nothingEndsTheDay(List<TimeRange> availableTimes, List<Event> eventList) {
+    int endOfDay = TimeRange.WHOLE_DAY.duration();
+
+    return (availableTimes.get(availableTimes.size() - 1).end() != endOfDay) &&
+      (eventList.get(eventList.size() - 1).getWhen().end() != endOfDay);
+  }
+
+  /**
+   * Check if the duration of our request fits between the end of the day and the end of the last event.
+   */ 
+  private static boolean fitsOnlyAtTheEnd(MeetingRequest request, List<TimeRange> availableTimes,
+    List<Event> eventList) {
+    return request.getDuration() <= TimeRange.END_OF_DAY - eventList.get(eventList.size() - 1)
+      .getWhen().end();
+  }
+
+  private static List getEventsWithRequestAttendees(MeetingRequest request, Collection<Event> events) {
+    List<Event> eventList = new ArrayList<>();
+    for (Event event : events) {
+      if (requestHasEventAttendees(request, event)) {
+        eventList.add(event);
+      }
+    }
+
+    return eventList;
+  }
+
+  private static boolean eventContainedOrOverlapped(TimeRange currEventTimeRange,
+    TimeRange nextEventTimeRange){
+    return currEventTimeRange.overlaps(nextEventTimeRange);
+  }
+
+  /** 
+   * Search for the next event that is not contained or overlapped and connect them.
+   */
+  private static void searchForNextEvent(List<Event> eventList, TimeRange currEventTimeRange,
+    int currEventPosition, List<TimeRange> availableTimes, MeetingRequest request) {
+    
+    // We already know that our next event is overlapped or contained, so start from position + 2 
```

Comment:
nit: you don't need to mention "overlapped or contained"... "contained" already implies in overlapping, so you don't need to mention it